Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+  - stringr: <https://github.com/tidyverse/stringr>
+  
+Some maintainers forget to list this URL, even though the package is developed in a public repository, but you still might be able to discover it via search.
+    
+Even if a package is not developed on a public platform, you can visit its source in the unofficial, read-only mirror maintained by [METACRAN](https://r-pkg.org/about). Examples:
+
+  - MASS: <https://github.com/cran/MASS>
+  - car: <https://github.com/cran/car>
+
+### Bundled package
+
+A __bundled__ package is a package that's been compressed into a single file. By convention (from Linux), package bundles in R use the extension `.tar.gz` and are sometimes referred to as "source tarballs". This means that multiple files have been reduced to a single file (`.tar`) and then compressed using gzip (`.gz`). While a bundle is not that useful on its own, it's a useful platform-agnostic intermediary between the other states.
+
+In the rare case that you need to make a bundle from a package you're developing locally, use `devtools::build()`. Under the hood, this calls `pkgbuild::build()` and, ultimately, `R CMD build`, which is described further in the [Building package tarballs][wre-tarball] secion of [Writing R Extensions][wre].
+
+Every CRAN package is available in this form, via the "Package source" field of individual landing pages. Continuing our examples from above, you could download the bundles `forcats_0.4.0.tar.gz` and `stringr_1.4.0.tar.gz` (or whatever the current versions may be). You could unpack such a bundle in the shell (not the R console) like so:
+
+```bash
+tar xvf forcats_0.4.0.tar.gz
+```
+
+If you decompress a bundle, you'll see it looks almost the same as a source package. This diagram summarises the files present in the top-level directory for source, bundled, and binary versions of devtools.
+
+*TODO: Remake this, possibly with a different (fictional or deliberately crafted?) package. Things I want to address: README.Rmd, .github/, CI config, pkgdown, LICENSE.md, data-raw/, updated treatment of NEWS.md, existence of vignettes/ in tarball, presence of build/ in tarball, absence of README.md in binary form.*
+
+```{r, echo = FALSE}
+knitr::include_graphics("diagrams/package-files.png")
+```
+
+The main differences between a source package and an uncompressed bundle are:
+
+* Vignettes have been built, so rendered outputs, such as HTML, appear below
+  `inst/doc/` and a vignette index appears in the `build/` directory, usually
+  alongside a PDF package manual.
+
+* A local source package might contain temporary files used to save time during
+  development, like compilation artefacts in `src/`. These are never found in 
+  a bundle.
+
+* Any files listed in `.Rbuildignore` are not included in the bundle. These are
+  typically files that facilitate your development process, but that should be
+  excluded from the distributed product.
+
+#### `.Rbuildignore` {#rbuildignore}
```

Comment:
As I worked on this, it became clear it's a pretty important topic and deserved a sub-header for link-ability.